src/Compilers/CSharp/Test/Semantic/Semantics/OverloadResolutionTests.cs

```diff
@@ -10853,7 +10853,7 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "Method").WithArguments("Program.Method<T>(in (T arg1, T arg2))").WithLocation(15, 9),
                 // (16,16): error CS1503: Argument 1: cannot convert from '(<null>, int)' to 'in (int arg1, int arg2)'
                 //         Method((null, 1));
-                Diagnostic(ErrorCode.ERR_BadArgType, "(null, 1)").WithArguments("1", "(<null>, int)", "in (int arg1, int arg2)").WithLocation(16, 16),
+                Diagnostic(ErrorCode.ERR_BadArgType, "(null, 1)").WithArguments("1", $"({CSharpResources.IDS_NULL}, int)", "in (int arg1, int arg2)").WithLocation(16, 16),
```

Comment:
>{CSharpResources.IDS_NULL} [](start = 88, length = 26)

I think it would be better to force english culture when we get the arguments.

Reply:
@sharwell provided a [UseCulture] attribute for those cases. It's a matter of preference. You might discuss this with the other teams to keep it consistent across the code base. Up until now we have not used the attribute but made the tests working in other cultures.

Reply:
@AlekseyTs The general preference has been to use localized resources "to the extent reasonably possible". Typically the point where we start looking for alternatives is when simple use of `string.Format` + a resource is not enough to get the test to pass consistently.

Reply:
> Up until now we have not used the attribute but made the tests working in other cultures.

This is not what I am suggesting. I am suggesting that the test helper that compares the diagnostics should request arguments for specific culture. All arguments for diagnostics should be localizable. No strings should be pulled from resources until someone asked for a message and then the requested culture should apply to everything in it.

Reply:
@sharwell 

> The general preference has been to use localized resources "to the extent reasonably possible". 

That is not the case for compiler tests. Compiler tests do not test localizability. Our approach is to make them culture independent and baselines human readable. The culture independence is achieved by explicitly requesting error messages for english culture. If that doesn't work, then this is an indication that the test infrastructure is broken, or the product is broken. The fallback to accessing resources explicitly is the last resort approach and I am not convinced it should be used here.

Reply:
The diagnostic is created [here](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs#L991) and the argument is created via  `argument.Display` 
The problem here is that the [BoundTupleExpression.Display](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/BoundTree/Formatting.cs#L85) uses a stringbuilder. So the argument is a text as opposed to a `LocalizableErrorArgument`. I don't see any straight forward way on how to resolve this. @AlekseyTs I would propose to revert my changes here and let the test fail on non english machines for now.

Reply:
> I would propose to revert my changes here and let the test fail on non english machines for now

It would be great if you could fix it because you can easily verify the effect, but it is up to you.
I think there is a simple fix for BoundTupleExpression.Display. Instead of building complete string, we can build a format string with argument markers and collect arguments in an array. Then we call to System.Runtime.CompilerServices.FormattableStringFactory.Create and return its result.

There is also an issue with 
```
    internal partial class BoundStackAllocArrayCreation
    {
        public override object Display
        {
            get { return string.Format(MessageID.IDS_StackAllocExpression.Localize().ToString(), ElementType, Count.Syntax); }
        }
    }
```

This one can be fixed by creating an object that is very similar to LocalizableErrorArgument, that calls Format after getting the string for requested culture.